admin/class-aioseop-notices.php

```diff
@@ -0,0 +1,714 @@
+				'time_start'     => time(),
+				'time_set'       => time(),
+			);
+		}
+
+		/**
+		 * Action Options Default Values
+		 *
+		 * Returns the default value for action_options in self::notices[$slug]['action_options'].
+		 *
+		 * @since 3.0
+		 *
+		 * @return array Action_Options variable in self::notices[$slug]['action_options'].
+		 */
+		public function action_options_defaults() {
+			return array(
+				'time'    => 0,
+				'text'    => __( 'Dismiss', 'all-in-one-seo-pack' ),
+				'link'    => '#',
+				'dismiss' => true,
+				'class'   => '',
+			);
+		}
+
+		/**
+		 * Set Notice Action Options
+		 *
+		 * Sets the Action Options in a Notice.
+		 *
+		 * @since 3.0
+		 * @access private
+		 *
+		 * @see self::insert_notice()
+		 * @see self::update_notice()
+		 *
+		 * @param array $action_options New action options to be added/updated.
+		 * @return array Action Options with new values added to old.
+		 */
+		private function set_action_options( $action_options ) {
+			$rtn_action_options = array();
+			// This helps prevent invalid notices, and empty arrays need to skip this operation when
+			// there is no actions intended for notice.
+			if ( ! is_array( $action_options ) ) {
+				$rtn_action_options[] = $this->action_options_defaults();
+				return $rtn_action_options;
+			}
+
+			foreach ( $action_options as $action_option ) {
+				$tmp_action_o = $this->action_options_defaults();
+
+				// For readability and tracking, refrane from using another Foreach loop with the array indexes.
+				// Button Delay Time.
+				$tmp_action_o['time'] = $this->default_dismiss_delay;
+				if ( isset( $action_option['time'] ) ) {
+					$tmp_action_o['time'] = $action_option['time'];
+				}
+
+				// Button Text.
+				if ( isset( $action_option['text'] ) && ! empty( $action_option['text'] ) ) {
+					$tmp_action_o['text'] = $action_option['text'];
+				}
+
+				// Link.
+				if ( isset( $action_option['link'] ) && ! empty( $action_option['link'] ) ) {
+					$tmp_action_o['link'] = $action_option['link'];
+				}
+
+				// Dismiss.
+				if ( isset( $action_option['dismiss'] ) ) {
+					$tmp_action_o['dismiss'] = $action_option['dismiss'];
+				}
+
+				// Class.
+				if ( isset( $action_option['class'] ) && ! empty( $action_option['class'] ) ) {
+					$tmp_action_o['class'] = $action_option['class'];
+				}
+
+				$rtn_action_options[] = $tmp_action_o;
+			}
+
+			return $rtn_action_options;
+		}
+
+		/**
+		 * Insert Notice
+		 *
+		 * Initial insert for a Notice and Activates it. Used strictly for adding notices
+		 * when no updating or modifications is intended.
+		 *
+		 * @since 3.0
+		 *
+		 * @uses self::activate_notice() Used to initialize a notice.
+		 *
+		 * @param array $notice See self::notices for more info.
+		 * @return boolean True on success.
+		 */
+		public function insert_notice( $notice = array() ) {
+			if ( empty( $notice['slug'] ) ) {
+				return false;
+			} elseif ( isset( $this->notices[ $notice['slug'] ] ) ) {
```

Comment:
could be in one line to make it simpler?

Reply:
Thanks, helped me find another issue in activate_notice